firebase-database/src/commonMain/kotlin/dev/gitlive/firebase/database/database.kt

```diff
@@ -32,13 +32,25 @@ expect fun Firebase.database(app: FirebaseApp): FirebaseDatabase
 expect fun Firebase.database(app: FirebaseApp, url: String): FirebaseDatabase
 
 expect class FirebaseDatabase {
+
+    class Settings {
+        val persistenceEnabled: Boolean
+        val persistenceCacheSizeBytes: Long?
+
+        companion object {
+            fun createSettings(persistenceEnabled: Boolean = false, persistenceCacheSizeBytes:  Long? = null): Settings
+        }
+    }
+
     fun reference(path: String): DatabaseReference
     fun reference(): DatabaseReference
-    fun setPersistenceEnabled(enabled: Boolean)
+    fun setSettings(settings: Settings)
```

Comment:
why not just    
`fun setSettings(persistenceEnabled: Boolean = false, persistenceCacheSizeBytes: Long? = null)`
What the point of making everyone write the boilerplate
`database.setSettings(FirebaseDatabase.Settings.createSettings(persistenceEnabled = true, persistenceCacheSizeBytes = 2048))`
?